tests/wan/lacp/test_wan_lag_min_link.py

```diff
@@ -0,0 +1,250 @@
+import time
+import pytest
+import random
+import logging
+import ipaddress
+import threading
+
+from queue import Queue
+from ptf import testutils, mask, packet
+from tests.common import config_reload
+from tests.common.utilities import wait_until
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.plugins.loganalyzer.loganalyzer import LogAnalyzer, LogAnalyzerError
+
+pytestmark = [
+    pytest.mark.topology('wan-4link'),
+    pytest.mark.device_type('vs')
+]
+
+
+@pytest.fixture(scope="module")
+def get_target_pcs(duthosts, enum_rand_one_per_hwsku_frontend_hostname, tbinfo):
+    duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
+    dut_mg_facts = duthost.get_extended_minigraph_facts(tbinfo)
+    # generate target pc tuples, be like:[("10.0.0.56", "10.0.0.57", "PortChannel0001")]
+    peer_ip_pc_pair = [(pc["addr"], pc["peer_addr"], pc["attachto"])
+                       for pc in dut_mg_facts["minigraph_portchannel_interfaces"]
+                       if ipaddress.ip_address(pc['peer_addr']).version == 4]
+
+    # generate pc tuples, fill in members,
+    # be like:[("10.0.0.56", "10.0.0.57", "PortChannel0001", ["Ethernet48", "Ethernet52"])]
+    all_pcs = [(pair[0], pair[1], pair[2], dut_mg_facts["minigraph_portchannels"][pair[2]]["members"])
+               for pair in peer_ip_pc_pair]
+    if len(all_pcs) < 2:
+        pytest.skip(
+            "Skip test due to no enough port channel in current topology, at least 2 port channels needed.")
+
+    # generate output pc tuples, fill in members,
+    # be like:[("10.0.0.56", "10.0.0.57", "PortChannel0001", ["Ethernet48", "Ethernet52"])]
+    out_pcs = [(pair[0], pair[1], pair[2], dut_mg_facts["minigraph_portchannels"][pair[2]]["members"])
+               for pair in all_pcs
+               if pair[2] in dut_mg_facts['minigraph_portchannels']
+               and len(dut_mg_facts["minigraph_portchannels"][pair[2]]["members"]) >= 2]
+    if len(out_pcs) < 1:
+        pytest.skip(
```

Comment:
Can you consider using the conditional mark function to skip the tests?
The conditional mark plugin is also able to skip based on minigraph facts.
The advantage is that the test can be skipped in collection stage.

Reply:
Add conditional maker in tests_mark_conditions.yaml to skip test when condition is not met.